build_product

```diff
@@ -1,16 +1,27 @@
 #!/bin/bash
 
 to_uppercase() {
-	printf '%s' "$1" | tr 'a-z' 'A-Z'
+	echo "$1" | tr '[:lower:]' '[:upper:]'
 }
 
 to_lowercase() {
-	printf '%s' "$1" | tr 'A-Z' 'a-z'
+	echo "$1" | tr '[:upper:]' '[:lower:]'
```

Comment:
Why to swap printf with echo? printf is a bit better, as it doesn't "swallow" (invalid) product names s.a. `-e`, `-n` etc.

Reply:
I will change back. I had changed to avoid the string format qualifier (as `printf "%s" <var>` is equivalent to `echo` sans the edge cases you describe and seems clearer for this use case) . Under mine, invalid product names still get detected and reported correctly:

```
The argument '-n' is not supported.
```

More concisely,

```bash
tr '[:upper:]' '[:lower:]' <<< "$1"
```

would work, but I'm guessing most people aren't familiar with the `<<<` operator.

Reply:
Yes, did not know what `<<<` does either. I would say that in our case, printf is just about 1% better than echo :-)